src/main/java/io/github/classgraph/MethodInfo.java

```diff
@@ -318,6 +324,24 @@ public String getTypeSignatureOrTypeDescriptorStr() {
         }
     }
 
+    public ClassInfoList getThrownExceptions() {
+        if (thrownExceptions == null && thrownExceptionNames != null) {
+            thrownExceptions = new ClassInfoList(thrownExceptionNames.length);
+            for (String thrownExceptionName : thrownExceptionNames) {
+                ClassInfo classInfo = scanResult.getClassInfo(thrownExceptionName);
+                thrownExceptions.add(classInfo);
```

Comment:
`classInfo` may be `null` here -- is it helpful to have a `null` placeholder in the list for classes that could not be found (because they were not discovered during the scan)?

Actually I forgot one more change you should make -- look at `extendScanningUpwards()` -- you should schedule the thrown exceptions for scanning in that method.

That raises the question though, is the `Exception` attribute always present, even if there is a generic `throws` type signature present?

Reply:
> That raises the question though, is the Exception attribute always present, even if there is a generic throws type signature present?

Yes it is. The `Exception` attribute will return the bound type though. So it just kind of works as is without having to strip parameterizations and such off of throws.

Reply:
> is it helpful to have a null placeholder in the list for classes

I'll remove null values if you feel like that's better.